src/androidTest/java/com/mixpanel/android/mpmetrics/MixpanelNotificationBuilderTest.java

```diff
@@ -200,6 +237,22 @@ public void testNoActionButtons() {
         verify(builderSpy, never()).addAction(any(Notification.Action.class));
     }
 
+    public void testValidNotificationBadge() {
+        final Intent intent = new Intent();
+        intent.putExtra("mp_message", "MESSAGE");
+        intent.putExtra("mp_badgecount", "2");
+        mpPushSpy.createNotification(intent);
+        verify(builderSpy).setNumber(2);
+    }
+
+    public void testInvalidNotificationBadge() {
+        final Intent intent = new Intent();
+        intent.putExtra("mp_message", "MESSAGE");
+        intent.putExtra("mp_badgecount", "0");
```

Comment:
what if the user actually wants to set the count to 0 (e.g. remove the badge)? What does the Android API do if you pass 0 to `setBadgeCount`?

More generally, if there are a number of different ways something can be invalid, let's have a test case for each one. E.g. `testNotificationBadgeCountLessThan0`, `testNotificationBadgeCountNotANumber` etc. This provides better test coverage and perhaps even more importantly, documents the edge cases we've identified and are explicitly handling.